hq/app/service/SecurityGroupsService.scala

```diff
@@ -0,0 +1,43 @@
+package service
+
+import akka.actor.ActorSystem
+import com.gu.Box
+import config.Config
+import model.{AwsAccount, SGInUse, SGOpenPortsDetail}
+import play.api.Configuration
+import rx.lang.scala.Observable
+import utils.attempt.{Attempt, FailedAttempt}
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+
+
+object SecurityGroupsService {
+
+  type EitherFlaggedSgs = Either[FailedAttempt, List[(SGOpenPortsDetail, Set[SGInUse])]]
+  type FlaggedSgs = List[(AwsAccount, EitherFlaggedSgs)]
+
+  private val sgsBox = Box[FlaggedSgs](List.empty)
```

Comment:
🤔 Thinking out loud:

For the securityGroupService, the ideal data structure of the Box (sgsBox) would be:
```
Box[Map[String, Attempt[Either[FailedAttempt, List[(SGOpenPortsDetail, Set[SGInUse])]]]]](Map.empty)
```

This would allow us to skip the filtering step in the controller (for single accounts) and minimise the amount transformation required